common/models/user.js

```diff
@@ -527,17 +533,15 @@ module.exports = function(User) {
         redirect: verifyOptions.redirect,
       });
 
-    verifyOptions.templateFn = verifyOptions.templateFn || createVerificationEmailBody;
-
-    // Email model
-    var Email =
-      verifyOptions.mailer || this.constructor.email || registry.getModelByType(loopback.Email);
+    verifyOptions.to = verifyOptions.to || user.email;
+    verifyOptions.subject = verifyOptions.subject || g.f('Thanks for Registering');
+    verifyOptions.headers = verifyOptions.headers || {};
 
-    // Set a default token generation function if one is not provided
-    var tokenGenerator = verifyOptions.generateVerificationToken || User.generateVerificationToken;
-    assert(typeof tokenGenerator === 'function', 'generateVerificationToken must be a function');
+    // assert the verifyOptions params that might have been badly defined
+    assertVerifyOptions(verifyOptions);
```

Comment:
verifyOptions assertion is now done right before generating token and going async